src/mscorlib/shared/System/IO/Path.Windows.cs

```diff
@@ -138,15 +141,29 @@ public static bool IsPathRooted(string path)
         // only contains whitespace characters an ArgumentException gets thrown.
         public static string GetPathRoot(string path)
         {
-            if (path == null) return null;
+            if (path == null)
+                return null;
+
             if (PathInternal.IsEffectivelyEmpty(path))
                 throw new ArgumentException(SR.Arg_PathEmpty, nameof(path));
 
-            // Need to return the normalized directory separator
-            path = PathInternal.NormalizeDirectorySeparators(path);
+            ReadOnlySpan<char> result = GetPathRoot(path.AsReadOnlySpan());
+            if (path.Length == result.Length)
+                return PathInternal.NormalizeDirectorySeparators(path);
+           
+            return PathInternal.NormalizeDirectorySeparators(new string(result));
```

Comment:
Should NormalizeDirectorySeparators start taking Span again to avoid the extra allocation?

Reply:
I don't think it will be that common that we end up needing to allocate twice. Most of the time the separators should already be canonicalized in the root. I'm fine with either not doing it or following up in another issue.

Reply:
we can do that. @JeremyKuhne should we revert back to the span overload ?

Reply:
I'd rather not put the span overload into this change for clarity if you want to close that rare extra alloc. I honestly don't expect it to happen often, so I'm totally ambivalent to making the extra effort, but I'd like to keep this change a little more focused.